owasp-top10-2017-apps/a9/cimentech/README.md

```diff
@@ -1,43 +1,132 @@
 
-After cloning this repository, you can type the following command to start the vulnerable application:
+```sh
+cd secDevLabs/owasp-top10-2017-apps/a9/cimentech
+```
 
 ```sh
 make install
 ```
 
-Then simply visit [localhost:80][App] !
+Then simply visit [localhost:80][App] ! 😆
+
+## Get to know the app 🏗
+
+To properly understand how this application works, you could:
+
+- Visit it's homepage!
+
+## Attack narrative
+
+Now that you know the purpose of this app, what could possibly go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or it's users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+### 👀
+
+#### Use of a vulnerable Drupal version allows for remote code execution
+
+
+It's possible to reach the server's web application from the standard HTTP port 80, as shown by the image below:
+
+<img src="images/attack1.png" align="center"/>
+
+Afterwards, by having a look at the `/robots.txt` file, it's possible to find the `CHANGELOG.txt` file in the `Disallow` field, as depicted by the image below:
+
+<img src="images/attack2.png" align="center"/>
+
+When accessed, an indication of the version of the Content Management System (Drupal) can be found, as shown below:
+
+<img src="images/attack3.png" align="center"/>
+
+Having the CMS version, it's possible to check on [exploit-db][3] if there are any exploits associated with that version, in this case, Drupal 7.57. The results of the search are depicted on the image below:
+
+<img src="images/attack4.png" align="center"/>
+
+By using [searchsploit](https://www.exploit-db.com/searchsploit), an attacker could also find this same result via terminal. To install it, simply type the following in your OSX terminal (keep in mind it might trigger your anti-virus software) :
+
+```sh
+brew install exploitdb
+```
+
+Then simply search for the version of the CMS found:
+
+```sh
+searchsploit drupal 7.
+```
+
+If you are using OSX, this command will help you to copy the exploit to your `/tmp` folder:
+
+```
+cp /usr/local/opt/exploitdb/share/exploit-database/exploits/php/webapps/44449.rb /tmp
+```
+
+## 🔥
+
+Running the malicious Ruby code, we have evidence that a remote code execution is possible on the web server, using the following commands as shown below:
+
+```sh
+ruby /tmp/44449.rb http://localhost
+```
+
+<img src="images/attack5.png" align="center"/>
+
+**NOTE**: You need to have Ruby installed on your system to run the exploit, for information on how to install it, click [here][1]!
+
+**NOTE 2**: If you are running Linux and came across an execution error when trying to run the exploit, please have a look at this [Issue][4] for information on how to proceed.
```

Comment:
I had the exact same issue running ruby on MacOS. I think this issue might happen in other systems as well. What do you think about just mentioning the issue, not necessarily Linux?

Reply:
Thanks for your feedback @Takehime! I'll add that!